flaml/automl.py

```diff
@@ -1085,14 +1128,25 @@ def custom_metric(
                 hyperparamter configurations for the corresponding estimators.
             seed: int or None, default=None | The random seed for np.random.
             **fit_kwargs: Other key word arguments to pass to fit() function of
-                the searched learners, such as sample_weight.
+                the searched learners, such as sample_weight. Include period and
+                freq as key word arguments for forecast task.
         '''
         self._start_time_flag = time.time()
         self._state.task = task
         self._state.log_training_metric = log_training_metric
         self._state.fit_kwargs = fit_kwargs
         self._state.weight_val = sample_weight_val
         self._state.groups = groups
+
+        if self._state.task == 'forecast':
+            if dataframe is not None:
+                if ('ds' not in dataframe) or ('y' not in dataframe):

```

Comment:
Is there a more transparent way to convey to the user the requirement on the dataframe's column names? There is a `label` argument for `fit()`, can we somehow use this? For example, we can have the `label` argument accepts a tuple of strings for the timestamps and label columns when the task is forecasting.

Reply:
Yes, I can do that